modules/ROOT/pages/dataweave-type-reuse-java-classes.adoc

```diff
@@ -0,0 +1,70 @@
+= Reusing Types from Java Classes
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+
+Use the import directive with the `java!` prefix when loading a Java class. The prefix `java!` tells DataWeave to use the Java module loader. When the module is loaded, you can call constructors, variables, and functions (see xref:dataweave-cookbook-java-methods.adoc[]). The class type is also available. DataWeave exposes the type of the Java class with the type name `Class`.
+
+Java classes are mapped to the DataWeave types listed in xref:dataweave-formats-java.adoc#java-value-mapping[Java Value Mapping].
+If a Java class is not present in the table, DataWeave treats the class as a `JavaBean` and maps it to an `Object` type with its fields being the properties name of the Java class and its corresponding mapped DataWeave type. In this case, DataWeave takes all properties from the Java getters.
```

Comment:
1. Is the class mapped to a DataWeave Object type? And the fields in that object match the names of the properties in the Java class? If so, could we say something like this?
In which case we can rephrase it as:
"... maps it to a DataWeave `Object` type. The fields in that object match the names of the properties in the Java class."

2. When we say, "In this case," what case are we referring to, and are we suggesting there are other cases? Are we talking about the example or something more general?

@TomoDC

Reply:
For 1. I would suggest something like "... maps it to a DataWeave Object type. The keys in that object match the names of the properties in the Java class and their values correspond to the DataWeave type that matches the property class."

For 2. Means the following example.